### DISCUSSION TITLE
Configuration help

### DISCUSSION THREAD

**Discussion by serbat** [slipx06/sunsynk-power-flow-card#238] [Huawei / iStore]
Good morning everyone

Meanwhile, congratulations for the really well done and very detailed card...

I wanted to ask for 2 clarifications if possible (maybe I wasn't able to)

I use a three-phase Huawei inverter and Wlcrs integration

1) The Wlcrs integration does not release the power values for the 2 strings used but only the total one, it could be useful in this case to use the voltage and ampere values of the 2 strings present to insert the values in the "PV1_power_186" and " PV2_power_187"

2) I noticed (but I'm surely wrong) that in the "essential_power" field the power value does not always appear, or rather, if I put the inverter power value when the inverter works the value appears, but when the inverter does not produce the power is at zero, because in this case it should read the power read by the power meter that takes from the network, in practice it should add the 2 values to always have a reading value

Thank you so much for your help, and sorry for my bad English

**Comment by slipx06** [comment]
@Roving-Ronin  has being doing a lot of work with the Huawei inverter and this integration. Have a look at this [discussion](https://github.com/slipx06/sunsynk-power-flow-card/issues/135) . Perhaps he can assist

**Reply by Roving-Ronin** [reply]
@serbat what is your Huawei system ?
ie 

1 or 2 inverters and model ? Assume a 5kW or 6kW M1 ?

Do you have a LUNA battery? If so 1 or 2 and attached to inverter 1? 5,10 or 15kWh capacity ?

Do you have a DTSU-666H (3p) meter ?

Of your interested (especially if you have a battery) I could send you the yaml for all the 'derived' sensors needed to be created from the WLCRS sensors to have what is needed to use with Sunsynk card.

Also where are you located and what type of electricity plan are you on?
ie:
- Flat Rate 24x7 with one price for grid import and exports?
- Time of Use (peak, shoulder, off peak) with different costs per period and an export rate ?

If you send me the plan details O can also see if I can use that to create another yaml example that takes all the WLCRS data and gives you sensors that tracks daily / monthly / yearly energy import and export and the cost for each.


As slipx06 mentioned I'm working on the Huawei 'how to', just slow going with spare time to confirm the different examples and add to documentation of examples.

**Reply by serbat** [reply]
HI

""1 or 2 inverters and model ? Hires at 5kW or 6kW M1 ?""
My inverter is the SUN2000 10K M1 model with 2 strings

""Do you have a LUNA battery? If so 1 or 2 and attached to inverter 1? 5.10 or 15kWh capacity ?""
Yes, I have 2 X 15Kwh modules for a total of 30Kwh

""Do you have a DTSU-666H (3p) meter ?""
Yes

""Time of Use (peak, shoulder, off peak) with different costs per period and an export rate ?""
In Italy we have 3 daily rates based on the time named F1 - F2 - F3

Very interesting and useful to have the details as you proposed, as far as the costs here are a problem, they change constantly and it makes little sense to manage the costs at this point, it is much better to know the relative consumption.

Your work is always very welcome and we end users are grateful for the time you dedicate to these projects, if you need further details or anything else let me know and I'll contribute if I can....

Grazie

**Reply by Roving-Ronin** [reply]
@serbat 2 x 15kW.... drool ;-) 
Can I ask what they cost in Italy, in Australia the WHOLESALE price for a 15kWh S0 is ~€7,100 plus install (& typically big margins atop that).

Personally waiting to see when and cost $ for when the LUNA S1 (7/14/21kWh) model becomes available, cost per kWh of those should equate to ~1/3rd the cost, given the claimed higher full cycles of the S1.

Can you give some more information on the power import / export prices though ? Do you just have different prices at different times or are you saying your like other parts of Europe with prices that fluctuate depending upon market demand / energy availability?

If the later is it one of those suppliers that has an integration for HA or publishes the current rate online (could screen scrape and feed that into a price sensor)?
Currently I've worked out a few differing prices tariffs so you end up with a sensor 'sensor.electricity_price' that has the current price published. Same for exporting to the grid and various prices or plans with $0.0x for first #kWh and thereafter $0.0y for the remainder exported that day.


BTW to get you going for the AUX entities in the card, do you have a backup generator or other power source to use there, or more likely do you have some power guzzlers you'd like to monitor? 

In the examples I'm doing (given users are probably geeks with lots of IT kit) I've been using those to monitor the Active Power of Servers and Network Gear. If you have something similar start giving some thought to what existing sensors for this gear you have, as we'll need to make some groups for them.

Same but for the Non_Essential entities, do you have say HVACs and EV charger (make/model?) that you want to track power and energy of?

Lastly for the Essential entities (under the house icon) do you have sensors like an IoTaWatt or Shelly Pro EM3 going you say Power use of all the light circuits and GPO (power points) in the house ?

J

**Reply by serbat** [reply]
Hi Roving-Ronin


Currently the 10K M1 inverter can be found from €1600 to €1850 plus shipping costs

Yes, unfortunately prices in Italy are currently highly variable, just for one example... in August 2023 the average price per kWh was €0.60, currently we are at €0.20, all of this due to a crisis in gas supply in Europe for the war in Ukraine....therefore, regarding energy prices, I believe that in Italy/Europe it has little meaning to calculate them

I'm not aware that Italian energy suppliers have anything that can be integrated into HA, obviously the data is there on their reference sites but I think it's complicated to manage it

No, at the moment I don't have an alternative energy generator although I had thought about it..

I believe that due to the way the Italian market is developing, PDCs (heat pumps... work with electricity) will certainly have more and more relevance.
In my case, having three-phase current, I structured the system with a current meter for each phase attributable to the 3 floors of the house at 220V, plus one dedicated to the 380V three-phase PDC
Obviously there are also household appliances such as dishwasher, washing machine, dryer, but these are not monitored at the moment.

Yes, I also have a charging station for a possible electric car, but I don't currently own it yet. The brand is Mennekes, but I can't remember the model now. It would also be interesting to have monitoring of this in the future.

for power detections I'm using, perfectly integrated with HA
https://catalogo.bticino.it/prodotto/soluzioni-per-edifici-residenziali-e-nel-terziario/living-now/BTI-F20T60A-IT

Saluti

**Reply by Roving-Ronin** [reply]
@serbat just wondering for your two strings, what is the total PV capacity / watts?  PDC = ?

How much are the batteries in Italy (is was referring to LUNA ESS for with " ~€7,100 plus install (& typically big margins atop that)").

Given you have two batteries, and the card (at least now) only supports 1 x inverter and 1 x battery being displayed, would you be OK with working on averaged SoC / volt / Amp / kW in&out (i.e. SoC 1 + SoC divided by 2 = reading) ?

@slipx06 really need a 'wide' card that has space to support having a couple of inverters and batteries ;-)

Re the groups I mention above, here's another users site below picture (2 x L1 inverters and 1 x LUNA battery) that I've setup remotely and have been using for testing the entities. As it shows, we use the AUX / Essential / Non-Essential to show IT kit power/energy and the same for Air Conditioner / EV charger / Lights and Powerpoints. If you don't have any monitoring on lights/powerpoint circuits or any smartplugs / Shelly PM monitoring individual devices ?? then assume the best we can do would be to use those to show Floor 1 / 2 / 3 active power and energy?


![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/108674933/7fb1e9ea-b0ef-45a2-87aa-1498552dac5b)

Also included some other cards that might be of interest (24hour overview is handy and the one below sunsynk is good for 'at a glance' view on things like mobile phone). Sunsynk is the best hands down though, with the level of detail for power flows :-)

**Reply by serbat** [reply]
Hi

My strings are made up of a total of 31 375W panels divided into 2 strings totaling 11625Kwh, while the PDC is 12Kwh max

The battery is made up of 2 modules of 15Kwh each and the inverter sees it as unique, I think it makes little sense to divide the SOC by 2, also because I have never seen significant differences in charge between the 2 modules battery, visual value given by the charging LEDs present on the shock of the battery module

Yes, I currently don't have any additional power meters but in the near future I will certainly add others divided in this way...
For each floor there are 3 separate power switches that manage the load of the general power sockets (therefore all the appliances on the floor), the general load of the lights and the air conditioners with heat pump, I don't think that in my case it is necessary to monitor each load would be too wasteful, at most I could think of interacting with the air conditioners according to the battery charge, that is, once the battery is 100% charged, start some air conditioners to improve the climate (something I have already done in the past but with little benefit )

So returning to the initial question, I think the best thing to do immediately is to show the power used divided by floor, plus the PDC together with the wallbox as these are the highest consumption levels in the building

This is currently my card... (sorry for the background but I'm connected remotely)


![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/02a611ba-e909-4dc5-abef-d33aecddd2f8)

Saluti

**Reply by Roving-Ronin** [reply]
Sorry PDC ? As in what does that stand for ?

Wall box ?? So you mean switch board for the whole house?

Can you send me the names of the sensors for the Active Power (W) of each of the floors / things to show and if they have a cumulative/lifetime Energy (kWh) sensor their names?

Re 'dividing by 2' I was pointing out on the sunsynk card we can only show ONE battery (if you want it all one one card instance). To achieve this I was saying we will need to create sensors that provide an average for the batteries. This is what we already do when we have multiple inverters, so instead of sensor.inverter_active_power and sensor.inverter_active_power_2 (being inverter #2) we use a new sensor 'sensor.inverters_active_power' and display that in the card.

Could you confirm the names of the sensors on battery #2, as I haven't seen any before. I assume they will be like: sensor.battery_state_of_charge_2 etc if like other sensors from wlcrs ?

**Reply by serbat** [reply]
Hi,
By PDC I mean the heat pump for home heating
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/88abdf09-b3a1-46e3-8444-d957c0e18eaf)

The wallbox is the electric car charging station
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/7d25f6d3-3af1-468c-8b89-969f7b0033f1)

These are the power meters currently in use
power meters on the 3 floors
Amperometro_comune  =  sensor.0x000474000085385c_power
Amperometro_gianna  =  sensor.0x0004740000827524_power
Amperometro_patty   =  sensor.0x0004740000926d8b_power


PDC power meters, one for each 380V phase
Amperometro_Templari_Fase1    =  sensor.0x0004740000a15eb2_power
Amperometro_Templari_Fase2    =  sensor.0x0004740000a15ec7_power
Amperometro_Templari_Fase3    =  sensor.0x0004740000a15f60_power

No, the battery is seen as a single sensor, the Huawei system up to 30Kw sees it as a single battery to have 2 battery sensors you must have a 30Kwh module (it is not possible to have one with a greater capacity) plus an additional module... only in this case will you have a second sensor, so I only have the "sensor.battery_state_of_charge" sensor

Saluti

**Reply by Roving-Ronin** [reply]
OK wow so your saying in HA with WLCRS your inverter only shows a single battery, that reports 30kWh size ?

I was going off the Huawei Modbus registers, that WLCRS uses/calls, as that has two different sets of registers (different numbers you query) to report all the data for batteries based upon battery (stack #1 of 3 x 5kWh modules for 15kWh total) and then totally seperate battery (stack #2).

What devices are you using that provides the PCD and floor sensors? Of If I understand your reply the actual sensors to call are the sensor.00x etc ? Wondering what gives such strange sensor names....

How are the PCD and ev charger linked into Home Assistant? Do they have an integration? Of your lucky they do and you can also call services in HA to control if / when / start / stop charging or using the PCD.

Also are you aware to get the temperature of the battery you need to modify the default (iirc) sensor.py file of WLCRS to add this sensor? Have you done this or do you need a copy of this (or how to modify, as this needs to be done when new version of WLCRS is updated)?

BTW have you also heard of EMHASS addon for HA, that allows controlling devices based upon power/solar availability etc... thinking it might be of interest to you also.

**Comment by serbat** [comment]
Hi,
I think you're right, checking on the Huawei application actually shows me 6 battery packs with the related soc data, temperature, charge, discharge, etc...
But honestly I consider it superfluous to monitor this data given that all management is sent back to the BSM of the battery itself and on which I don't think it is possible to intervene

if you notice in the previous message I indicated which sensors are for the various floors and which are for the PDC which, being three-phase current, are 3, one per phase.
For other sensors currently used, they are managed by another BPT home automation system and cannot be integrated into HA, and this is why I am trying to make HA work because it is much, much more flexible, little by little I will transfer everything to HA....

The names of the sensors reported are those read by the original Zigbee, then I renamed them but those are by default

For the moment PDC has not yet developed anything that can be integrated, I have spoken with some technicians from the company who are seeing what they can do, but it all depends on the requests they receive

No, I am not aware of the modification of the sensor.py file, I would be grateful if you can send me the necessary documentation... assuming I can understand it :)

**Reply by Roving-Ronin** [reply]
The battery pack registers are the individual 5kWh modules  (3 per 15 kWh battery... .so 6 in total for 2 batteries). 
In WLCRS these individual module temperatures have already been looked at and WLCRS does NOT include support for them. 

The only battery temperatures that WLCRS can (if you update that sensor.py file) support are for the overall battery stacks (i.e. battery_battery_1_temperature refers to 15kWh battery number 1) as reported by each BMS.

See this issue thread, in which I have posted looking at these: https://github.com/wlcrs/huawei_solar/discussions/613

As per this table, Huawei modbus register refers to the individual LUNA 15kW units, as 'Energy Storage Unit', this being the temperature overall that the BMS is reporting.

![302072308-affb8c80-fbdc-4971-bcd1-c7b2f3f07700](https://github.com/slipx06/sunsynk-power-flow-card/assets/108674933/960c9132-7ec0-49ed-897d-8b91e201ef79)


If you could update your sensor.py anyhow and let me know if you can then see a temperature sensor in Battery 1.  I may be disabled when you look in the battery screen and the list of sensors, if so you'll just need to open it and click on the switch for 'enable', then wait 30 seconds and see it start working.

See if your Battery 2 also have this sensor appear, and repeat enabling it.

**Comment by serbat** [comment]
Hi,
With us at the moment it is 2.52pm

Ok, this is your screen with the data you attached...
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/31070c4a-2c54-4bc4-bcf0-7a71c9b702c2)

this is the screen with the data that I managed to enter...
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/719c2a57-6e59-4c1a-aa2c-4f93aa445565)

**Reply by Roving-Ronin** [reply]
@serbat As your a 3 phase inverter / meter user, going to your Inverter can you please make sure you have these sensors enabled:

Phase A - Current
Phase B - Current
Phase C - Current

So it looks like this (plus PV2 - Current and PV2 - Voltage.  They are only disabled on my setup, as I only use 1 big string per inverter:
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/108674933/47fc68aa-0777-456a-bbe9-f11cb198235e)


Then on your Power Meter, enable   (If I have missed any, or any are already on by default please let me know !!  I need for documenting the steps for others):

Phase A - Active Power
Phase A - Current
Phase A - Voltage
Phase B - Active Power
Phase B - Current
Phase B - Voltage
Power Factor
Frequency

Power Meter should look like this:

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/108674933/3613606c-b166-4664-be92-56146ff44d73)


Could you also go into the Huawei integration screen and send your screenshot of the Huawei devices that WLCRS is detecting (like):

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/108674933/fdba2da8-eb02-4613-9346-2943972d1f01)

then assuming its showing a second battery, open up the second battery and select one of the sensors and tell me the entity ID of one of the senors? I'm needing to confirm the naming standard the entities on the second battery have.

For example Battery 1 - State of Capacity sensor entity ID is sensor.battery_state_of_capacity  I need to confirm is Battery 2 - State of Capacity sensor entity ID is sensor.battery_state_of_capacity_2

**Comment by Roving-Ronin** [comment]
If you don't have battery temperate, your supposed to make a backup of the /homeassistant/custom_components/huawei_solar/sensor.py file and then add the lines in green from here.

https://github.com/wlcrs/huawei_solar/compare/c023168..user-defined-extra-sensors

If you want to cheat, I've created a forked respository with the modified sensor.py that you can copy and paste, at:
https://github.com/Roving-Ronin/huawei_solar/blob/main/sensor.py

Also can you update this:
day_aux_energy: sensor.sunsynk_power_flow_aux_devices_energy    to:    day_aux_energy: sensor.sunsynk_power_flow_aux_devices_energy_daily

This is for a group that we'll get around to creating soon, but just noticed I had a typo there.

**Comment by serbat** [comment]
Ok, congratulations you are doing a very complex and complete job....
Over the weekend I'll try to make the changes you suggested, for me they're a little complicated but I can do it....

However, it remains my priority to complete the entire flow of energy produced and imported, then obviously all these small details...

**Comment by Roving-Ronin** [comment]
Is your EV charger an Mennekes Amtron Xtra ? 
If so do you have it connected to your home network (preferably with a fixed IP) 

If so, you'll be able to add it into HA and then directly pull its data for use in the sunsynk card.

**Comment by serbat** [comment]
Hi,
these are my sensors

Inverter
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/b7252cff-f5ad-4e1e-b320-88683ca1359a)

Power meter
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/96853828-f5d6-4b69-8dc3-daf0f6476211)

and these are the detected equipment (there are also 31 optimizers)
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/29bba8b5-3ed2-4684-9fd9-ac2d73a23e43)
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/5d8bd154-5884-47c5-83e4-d70a6dbd27e4)

Unfortunately my wallbox doesn't have an internet connection
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/9843c231-617e-4242-a33d-99a52f23a929)

**Reply by Roving-Ronin** [reply]
@serbat  OK so please enable the sensors mentioned above, then see if they appear on the Sunsynk card.

Looking at your power meter, it looks like you have 3 sensors labelled 'current' (Corrente), that you'll need to open up and check the entityID to identify what phase they are for. Once identified I would suggest you then update the 'Name' field from Corrente to 'Corrente di Fase A/B/C etc'

**Following that make sure you have enabled.**

**POWER METER**
- Corrente Fase A
- Corrente Fase B
- Corrente Fase C
- Tensione di Fase A
- Tensione di Fase B
- Tensione di Fase C

**INVERTER**
- Corrente Fase A
- Corrente Fase B
- Corrente Fase C
- Fattore di potenza
- Internal Temperature

Can you then send me screenshot of the card, that should now be showing the sensors (highlighted in red).

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/108674933/36c861ed-cceb-4986-88ad-43a64e52bcdd)

Also you might want to replace the 'grid_name: Your-Grid-Name'  with your electricity retailer name, or delete that line to default to showing 'Grid'.

Also when you battery is charged (more mostly at least) can you send me a screenshot of its 'Sensors' and 'Configuration' panels from HA? I have a strong suspicion your second battery is not connected to the RS485 communications (or not correctly wired). As WLCRS should be showing two seperate battery devices in HA.

**Comment by serbat** [comment]
Hi,

ok, once the sensors were activated the names corrected themselves
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/f9b66721-02e8-4e44-b91d-650a3772600c)

this is the new view of the card... consider that here it is 7 in the morning and the sun hasn't risen yet
![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/274f1c60-9953-40ad-8834-1719ae402fe7)

**Comment by serbat** [comment]
Hi,

This is the current situation, as you will notice the power value is missing in circle 1.
I wasn't able to make it work correctly, I can display either the power from the inverter or the power from the meter. I believe that in this bubble there should be the sum of the 2 values

In bubble 2 and 3 the respective values of PV1 (560.8X1) and PV2 (569X2.2) should appear, consequently in bubble 4 the sum of bubbles 2+3 should appear.

Further question, where is the reference .yaml file for this card?

![immagine](https://github.com/slipx06/sunsynk-power-flow-card/assets/83161647/f1ed530d-91cd-4639-b95a-f100d1e46974)

**Reply by Roving-Ronin** [reply]
Hi @serbat 

Yes I have the additional yaml files, that you'll have to place in the /homeassistant/packages directory, and that will also need to be modified to use the sensors for your floors / PDC etc.

Before then however we need to sort out the actions I have flagged , especially working out why you only have one battery appear. As flagged before if you have 2 x 15kWh battery, they should appear as 2 units with WLCRS.

If you tried to use these yaml files now, HA will put out a yaml error and not start your HA, hence we need to work out why the 2nd battery is missing.

Apart from sending me the screenshot when the battery is near full, how about in Huawei FusionSolar application.... does it also show the two batteries?
If not then we'll need to look at the rs485 communication links and ensure the other battery is wired into the rs485 daisy chain / or if a wire has become lose.

From memory in the monitoring view of FusionSolar portal (on a PC) you should see the name of your Site, of your click on that it should then show the sDongle your using to connect it to the network, expand that and it will show the inverter and then under the inverter it should show 
- Optimizers (31)
- Power Meter
- Battery(s)

If you're using the FusionSolar app on a mobile phone, go to the last tab at the bottom (devices) and it should show you dongle and inverter. At the bottom of the inverter it should say 'connected devices', if you click that it should show you:

- DTSU-666H
- Optimizers (31)
- Battery (2)

**Comment by samumar82** [comment]
HEllo, trying fo figure it out but stuck at this point:
![Screenshot 2024-04-02 alle 11 12 19](https://github.com/slipx06/sunsynk-power-flow-card/assets/160247359/5a96a1c6-0860-4576-9686-c8239c721fcd)
Anyone can help me figure out why I'm not able to set A-B-C correctly?
Need to have the daily split for both energy imported from the grid and exported. No lifecycle timeframe but just daily. Same fot the total PV production and house actual consumption & total daily consumption

**Reply by Roving-Ronin** [reply]
@samumar82  Are you using the card yaml examples and sensor yaml files for the Huawei inverters? FYI I have just sent a 'pull' to merge some updates of these into the documentation/examples. Keep an eye open for that update and please use the updated files.
